dev-docs/security-overview.md

```diff
@@ -0,0 +1,74 @@
+
+Next, the CLI the connects to the Kubernetes API server (kube-apiserver) using the kubeconfig file it received from the Bootstrapper. This results in an mTLS connection between the CLI and the Kubernetes API server. The CLI uses this connection for two essential operations at the Kubernetes level:
+1. It writes the runtime measurements of the applicable node image to a specific key in the cluster's etcd store.
+2. It executes the [hardcoded Helm charts](#cli-root-of-trust) of the three Constellation services KeyService, JoinService, and VerificationService.
+
+The latter causes the first node to download, verify, and run these three services. The services' containers are hosted at https://ghcr.io/edgelesssys.
+
+After this, the Constellation cluster is operational on the first node.
+
+## Cluster growth
+
+Additional nodes can now join the cluster - either as control-plane nodes or as worker nodes. For both, the process for joining the cluster is identical. First, the Bootstrapper running on a *new node* contacts the JoinService of the existing cluster. The JoinService verifies the remote-attestation statement of the new node using the runtime measurements stored in etcd. On success, an aTLS connection between the two is created, which is used mainly for the following (see the the [interface definition](https://github.com/edgelesssys/constellation/blob/main/joinservice/joinproto/join.proto) for a comprehensive list of exchanged data):
+1. The new node sends a certificate signing request (CSR) to the JoinService.
+1. The JoinService issues a corresponding certificate and sends it to the new node. The JoinService uses the signing key of the Kubernetes cluster for this, which is [generated by kubeadm](https://kubernetes.io/docs/setup/best-practices/certificates/).
+1. The JoinService sends a [kubeadm token](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-token/) to the new node.
+1. The JoinService sends the hashes of the to-be-installed Kubernetes binaries to the new node.
+1. The JoinService sends encryption key for the new node's local storage. This key is generated and managed by the cluster's KeyService.
+1. The JoinService sends the certificate of the cluster's control plane to the new node.
+
+After this, the aTLS connection is closed and the node is marked as "initialized". 
+
+The Bootstrapper on the new node uses the kubeadm token to download the configuration of the cluster from the Kubernetes API server. Subsequently, it downloads, verifies, and runs the given Kubernetes binaries accordingly. The kubeadm token is never used after this. 

```

Comment:
> download the configuration of the cluster from the Kubernetes API server. Subsequently, it downloads, verifies, and runs the given Kubernetes binaries accordingly

That's not correct. The components are part of the `IssueJoinTicketResponse`: https://github.com/edgelesssys/constellation/blob/fix/openstack/set-region-in-cloud-client/joinservice/joinproto/join.proto#L47